db/mysql-instance.yml

```diff
@@ -162,35 +173,41 @@ Resources:
     DeletionPolicy: Retain
     UpdateReplacePolicy: Retain
     Properties:
-      DeletionProtection: true
-      # CloudFormation cannot update a stack when a custom-named resource requires replacing,
-      # so when replacement is required (such as when restoring from snapshot),
-      # use a new instance name such as oldname-restoredYYYYMMDD
-      # Better yet to self manage tearing down of the existing resource, use a new stack with a different instance name
+      AllocatedStorage: !Ref DBAllocatedStorage
+      AllowMajorVersionUpgrade: false
+      AutoMinorVersionUpgrade: false
+      BackupRetentionPeriod: !Ref BackupRetentionPeriod
+      CopyTagsToSnapshot: true
+      DBInstanceClass: !Ref DBInstanceClass
       DBInstanceIdentifier:
         !If [NamedInstance, !Ref DBInstance, !Ref "AWS::NoValue"]
       DBName: !If [CreateDB, !Ref CreateDBName, !Ref "AWS::NoValue"]
-      AllocatedStorage: !Ref DBAllocatedStorage
-      StorageType: gp2
-      DBInstanceClass: !Ref DBInstanceClass
       DBSnapshotIdentifier:
         !If [UseSnapshot, !Ref DBSnapshot, !Ref "AWS::NoValue"]
+      DBSubnetGroupName: !Sub Platform-${EnvironmentType}-${AWS::StackName}-db
+      DeleteAutomatedBackups: true
+      DeletionProtection: true
+      EnablePerformanceInsights: !If [IsProduction, true, false]
       Engine: MySQL
       EngineVersion: !Ref EngineVersion
       MasterUsername: !If [UseSnapshot, !Ref "AWS::NoValue", !Ref DBUser]
       MasterUserPassword:
         !If [UseSnapshot, !Ref "AWS::NoValue", !Ref DBPassword]
-      MultiAZ: false
+      MultiAZ: !If [IsProduction, true, false]
+      PerformanceInsightsRetentionPeriod: !If [IsProduction, 7, !Ref "AWS::NoValue"]
+      PreferredBackupWindow: "08:00-08:30"
+      PreferredMaintenanceWindow: "tue:06:15-tue:06:45"
+      PubliclyAccessible: true
```

Comment:
This also presents a potentially significant change for updating resources from the previous version of the template